src/actions/collectiblesActions.js

```diff
@@ -0,0 +1,136 @@
+      const {
+        token_id: id,
+        asset_contract: assetContract,
+        name,
+        description,
+        image_preview_url: image,
+      } = collectible;
+      const { name: category, address: contractAddress } = assetContract;
+      const collectibleName = name || `${category} ${id}`;
+      const assetData = {
+        id,
+        category,
+        name: collectibleName,
+        description,
+        icon: (/\.(png)$/i).test(image) ? image : '',
+        contractAddress,
+        assetContract: category,
+        tokenType: COLLECTIBLES,
+      };
+      collectibles.push(assetData);
+    });
+
+    dispatch(saveDbAction('collectibles', { collectibles }, true));
+    dispatch({ type: UPDATE_COLLECTIBLES, payload: collectibles });
+  };
+};
+
+export const fetchCollectiblesHistoryAction = () => {
+  return async (dispatch: Function, getState: Function, api: Object) => {
+    const { wallet: { data: wallet } } = getState();
+    const collectiblesHistory = [];
+    const transactionEventsResponse = await api.fetchCollectiblesTransactionHistory(wallet.address);
+
+    if (transactionEventsResponse.error || transactionEventsResponse.asset_events === undefined) return;
+
+    transactionEventsResponse.asset_events.forEach((event) => {
+      const {
+        asset,
+        transaction,
+        to_account: toAcc,
+        from_account: fromAcc,
+      } = event;
+      const {
+        asset_contract: assetContract,
+        name,
+        token_id: id,
+        description,
+        image_preview_url: image,
+      } = asset;
+      const { name: category, address: contractAddress } = assetContract;
+      const { transaction_hash: trxHash, block_number: blockNumber, timestamp } = transaction;
+
+      const collectibleName = name || `${category} ${id}`;
+
+      const assetData = {
+        id,
+        category,
+        name: collectibleName,
+        description,
+        icon: (/\.(png)$/i).test(image) ? image : '',
+        contractAddress,
+        assetContract: category,
+        tokenType: COLLECTIBLES,
+      };
+
+      const transactionEvent = {
+        to: toAcc.address,
+        from: fromAcc.address,
+        hash: trxHash,
+        createdAt: (new Date(timestamp).getTime()) / 1000,
+        _id: transaction.id,
+        protocol: 'Ethereum',
+        asset: collectibleName,
+        contractAddress,
+        value: 1,
+        blockNumber,
+        status: 'confirmed',
+        type: COLLECTIBLE_TRANSACTION,
+        icon: asset.image_preview_url,
+        assetData,
+      };
+
+      collectiblesHistory.push(transactionEvent);
+    });
+    dispatch(getExistingTxNotesAction());
+    dispatch(saveDbAction('collectiblesHistory', { collectiblesHistory }, true));
+    dispatch({ type: SET_COLLECTIBLES_TRANSACTION_HISTORY, payload: collectiblesHistory });
+  };
+};
+
+export const fetchAllCollectiblesDataAction = () => {
+  return async (dispatch: Function) => {
+    dispatch(fetchCollectiblesAction())
```

Comment:
```suggestion
    await dispatch(fetchCollectiblesAction());
    await dispatch(fetchCollectiblesHistoryAction());
```

Reply:
Will include that in upcoming push with updates. Thanks! :)